manifests/web.pp

```diff
@@ -400,6 +402,15 @@
       $directory_deny = { 'deny' => 'from all', 'order' => 'Deny,Allow', }
     }
 
+    $location_api_access = $zabbix_api_access ? {
+      undef   => $directory_allow,
+      default => if versioncmp($apache::apache_version, '2.4') >= 0 {
```

Comment:
I'm wondering if we really need to support apache 2.2. dropping this logic makes the whole file way easier to read.

Reply:
Yeah, a separate commit to drop that makes sense I think.  After all, are there any supported OSes that still use Apache 2.2??
If this was zabbix-agent code then I'm all for being more conservative in what we drop, but for the server side stuff, not so much.

FWIW, I had a think about [constructs](https://gist.github.com/alexjfisher/eeda5a8d54f9895f37b7b3aafea54d6e) that might be more readable.  Conclusion was they're all pretty horrible!

Reply:
Do you think about removing entirely the `$directory_allow` / `$directory_deny` variables, thus removing an entire `merge()`, like so:
```diff
-        merge(
-          merge({
-              path     => '/usr/share/zabbix',
-              provider => 'directory',
-          }, $directory_allow),
-          $fcgi_filematch
+        merge({
+            path     => '/usr/share/zabbix',
+            provider => 'directory',
+            require  => 'all granted',
+          }, $fcgi_filematch

```

Or should I keep the variables, just remove the support for old apache like so:
```diff
-    # Check which version of Apache we're using
-    if versioncmp($apache::apache_version, '2.4') >= 0 {
-      $directory_allow = { 'require' => 'all granted', }
-      $directory_deny = { 'require' => 'all denied', }
-    } else {
-      $directory_allow = { 'allow' => 'from all', 'order' => 'Allow,Deny', }
-      $directory_deny = { 'deny' => 'from all', 'order' => 'Deny,Allow', }
-    }
-
+    $directory_allow = { 'require' => 'all granted', }
+    $directory_deny = { 'require' => 'all denied', }
     $location_api_access = $zabbix_api_access ? {
       undef   => $directory_allow,
-      default => if versioncmp($apache::apache_version, '2.4') >= 0 {
-        { 'require' => $zabbix_api_access.map |$host| { "host ${host}" }, }
-      } else {
-        { 'allow' => $zabbix_api_access, 'order' => 'Allow,Deny' }
-      }
+      default => { 'require' => $zabbix_api_access.map |$host| { "host ${host}" }},
     }
```

And keep the mess down?
```
        merge(
          merge({
              path     => '/usr/share/zabbix',
              provider => 'directory',
          }, $directory_allow),
          $fcgi_filematch
```

Reply:
I did option 1, but tell me if you think it should be done otherwise.

Reply:
Aslo, I added an explicit `fail()` for apache < 2.4